Add support for secret chats in SimpleEventHandler!

mtalaeii opened the pull request:
- Add SecretChat service message classes.
- Separate PrivateMessage classes.
-  Add support from decreyptedMedias in secret chats.

mtalaeii committed:
Add support for secret chats in SimpleEventHandler!
- Add SecretChat service message classes.
- Separate PrivateMessage classes.
- Note that chatId and senderId marked as TODO to be added soon ...
- cs-fix

mtalaeii committed:
Merge remote-tracking branch 'origin/v8' into v8

mtalaeii committed:
More fixes and works on Media classes!

mtalaeii committed:
Merge branch 'danog:v8' into v8

danog commented on src/EventHandler/Media/Document.php:
```diff
@@ -19,18 +19,42 @@
+    /** Thumbnail width only for secret chats. */
+    public readonly int $thumbWidth;
+    /** Caption only for secret chats. */
+    public readonly ?string $caption;
```
The caption is not only for secret chats

danog replied:
You should remove this field altogether, use message instead

danog commented on src/EventHandler/Media/Photo.php:
```diff
@@ -19,22 +19,42 @@
 {
     /** If true; the current media has attached mask stickers. */
-    public readonly bool $hasStickers;
+    public readonly ?bool $hasStickers;
```
not nullable

mtalaeii replied:
We dont have it in secret chats so media is object of secret chat photos its can be null

danog replied:
The absence of a field implies that a photo can never have stickers, just do ?? false below

danog commented on src/EventHandler/Media/Photo.php:
```diff
@@ -19,22 +19,42 @@
     ) {
         parent::__construct($API, $rawMedia, $protected);
-        $this->hasStickers = $rawMedia['photo']['has_stickers'];
+        $this->hasStickers = $rawMedia['photo']['has_stickers'] ?? null;
```
?? false

danog commented on src/EventHandler/Media/Document.php:
```diff
@@ -19,18 +19,42 @@
 final class Document extends Media
 {
+    /** Content of thumbnail file (JPEGfile, quality 55, set in a square 90x90) only for secret chats. */
+    public readonly string $thumb;
```
Nullable, and it's probably best to actually use Bytes here

danog commented on src/EventHandler/Media/Document.php:
```diff
@@ -19,18 +19,42 @@
+    /** Content of thumbnail file (JPEGfile, quality 55, set in a square 90x90) only for secret chats. */
+    public readonly string $thumb;
+    /** Thumbnail height only for secret chats. */
+    public readonly int $thumbHeight;
```
Nullable

danog commented on src/EventHandler/Media/Document.php:
```diff
@@ -19,18 +19,42 @@
+    /** Thumbnail height only for secret chats. */
+    public readonly int $thumbHeight;
+    /** Thumbnail width only for secret chats. */
+    public readonly int $thumbWidth;
```
Nullable

danog commented on src/EventHandler/Media/Photo.php:
```diff
@@ -19,22 +19,42 @@
-    public readonly bool $hasStickers;
+    public readonly ?bool $hasStickers;
+    /** Content of thumbnail file (JPEGfile, quality 55, set in a square 90x90) only for secret chats. */
+    public readonly string $thumb;
```
Same as before

danog commented on src/EventHandler/Media/Video.php:
```diff
@@ -25,6 +26,12 @@ final class Video extends AbstractVideo
     /** If true; the current media has attached mask stickers. */
     public readonly bool $hasStickers;
+    /** Content of thumbnail file (JPEGfile, quality 55, set in a square 90x90) only for secret chats. */
+    public readonly string $thumb;
```
You get the idea :)

danog commented on src/EventHandler/Message/SecretMessage.php:
```diff
@@ -0,0 +1,41 @@
+    /** Random message ID of the message this message replies to (parameter added in layer 45) */
+    public readonly ?int $replyToRandomId;
+    /** Specifies the ID of the inline bot that generated the message (parameter added in layer 45) */
+    public readonly ?string $viaBotName;
```
Use viaBotId from Message instead, resolving the username with getId

danog commented on src/EventHandler/Message/SecretMessage.php:
```diff
@@ -0,0 +1,41 @@
+    /** Whether the webpage preview is disabled */
+    public readonly ?bool $noWebpage;
+    /** Random message ID of the message this message replies to (parameter added in layer 45) */
+    public readonly ?int $replyToRandomId;
```
getReply would also be neat

danog commented on src/EventHandler/Message/Service/SecretChat/ActionTyping.php:
```diff
@@ -0,0 +1,38 @@
+/**
+ * User is preparing a message: typing, recording, uploading, etc.
+ */
+class ActionTyping extends ServiceMessage
```
I think we should reuse the normal Typing service message

danog commented on src/MTProtoTools/FilesAbstraction.php:
```diff
@@ -58,18 +58,21 @@ trait FilesAbstraction
     public function wrapMedia(array $media, bool $protected = false): ?Media
     {
-        if ($media['_'] === 'messageMediaPhoto') {
+        if ($media['_'] === 'messageMediaPhoto' || $media['_'] == 'decryptedMessageMediaPhoto') {
```
I'd like to use the same objects for both secret and normal media tbh

mtalaeii committed:
More fixes and works on Media classes!

mtalaeii committed:
Merge branch 'v8' into v8

danog committed:
Merge branch 'v8' into secret

danog committed:
Fix

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Merge branch 'v8' into v8

mtalaeii committed:
Update secret chat and SecretMessage strategy
- Update some field in AbstractMessage and Message class
- Remove some Todos as it implemented!
- just 1 Todo in mean time and need changes thats getReply method
- Run composer cs-fix

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Fix

mtalaeii committed:
Merge remote-tracking branch 'origin/v8' into v8

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Add some fixes and stuff
- Force getSecretMessage method to return SecretMessage
- Implement last Todo (getReply method inside SecretMessage class)
- Add private file field into media class for supporting download medias from secret chats
- Run composer cs-fix

mtalaeii committed:
Final Fix

mtalaeii commented:
So this pull request is finally ready for merge as i requested for review before 
I cant create another review request please Review it :)
@danog

danog commented on src/EventHandler/AbstractMessage.php:
```diff
@@ -68,18 +69,20 @@ public function __construct(
+        $decryptedMessage = $this instanceof SecretMessage ? $rawMessage['decrypted_message'] : null;
+        $this->out = $rawMessage['out'] ?? false;
+        $this->id = $this instanceof SecretMessage ? $rawMessage['random_id'] : $rawMessage['id'];
+        $this->chatId = $this instanceof SecretMessage ? $this->getClient()->getSecretChat($rawMessage['chat_id'])->chatId :$info['bot_api_id'];
```
pls avoid using getSecretChat twice

danog commented on src/EventHandler/AbstractMessage.php:
```diff
@@ -68,18 +69,20 @@ public function __construct(
+        $this->mentioned = $rawMessage['mentioned'] ?? false;
+        $this->silent = $rawMessage['silent'] ?? $decryptedMessage['silent'];
+        $this->ttlPeriod = $rawMessage['ttl_period'] ?? $decryptedMessage['ttl'] ?? null;
+        if ($this instanceof SecretMessage && isset($decryptedMessage['reply_to_random_id'])) {
```
Put this instead of $this->replyToMsgId = null; in the last else of this function

danog commented on src/EventHandler/AbstractPrivateMessage.php:
```diff
@@ -0,0 +1,44 @@
+     * Notify the other user in a private chat that a screenshot of the chat was taken.
+     *
+     */
+    public function screenShot(): DialogScreenshotTaken
```
Make this abstract, instead

danog commented on src/EventHandler/Message/SecretMessage.php:
```diff
@@ -0,0 +1,67 @@
+    /** Whether the webpage preview is disabled */
+    public readonly ?bool $noWebpage;
+    /** Random message ID of the message this message replies to (parameter added in layer 45) */
+    public readonly ?int $replyToRandomId;
```
Remove this

danog commented on src/EventHandler/Message/Service/SecretChat/ActionDeleteMessages.php:
Can your please move this out of the secret namespace, I want this to be used for normal chats too (the only thing that changes is the ID anyway, also rename randomIds to ids)

danog commented on src/EventHandler/Message/Service/SecretChat/ActionFlushHistory.php:
Can your please move this out of the secret namespace, I want this to be used for normal chats too (the only thing that changes is the ID anyway, also rename randomIds to ids)

danog commented on src/EventHandler/Message/Service/SecretChat/ActionReadMessages.php:
Can your please move this out of the secret namespace, I want this to be used for normal chats too (the only thing that changes is the ID anyway, also rename randomIds to ids)

danog commented on src/EventHandler/Message/Service/SecretChat/ActionScreenshotMessages.php:
Can your please move this out of the secret namespace, I want this to be used for normal chats too (the only thing that changes is the ID anyway, also rename randomIds to ids)

danog commented on src/EventHandler/Message/Service/SecretChat/ActionSetMessageTTL.php:
Can your please move this out of the secret namespace, I want this to be used for normal chats too (the only thing that changes is the ID anyway, also rename randomIds to ids)

danog commented on src/EventHandler/Message/Service/SecretChat/SecretUserTyping.php:
Pls reuse normal actions

danog commented on src/MTProtoTools/UpdateHandler.php:
Don't forget to handle updateNewOutgoingEncryptedMessage

danog commented on src/MTProtoTools/FilesAbstraction.php:
??? This conflicts with the definition in Files.php

danog commented on src/MTProtoTools/Files.php:
Please always use === or !===

danog commented on src/MTProtoTools/Files.php:
```diff
@@ -77,15 +77,21 @@ trait Files
+        if ($media['_'] !== 'messageMediaDocument' && $media['_'] !== 'decryptedMessageMediaDocument' && $media['_'] !== 'decryptedMessageMediaExternalDocument') {
             return null;
         }
+        if ($media['_'] == 'decryptedMessageMediaDocument' || $media['_'] == 'decryptedMessageMediaExternalDocument') {
```
??

danog commented on src/MTProtoTools/Files.php:
```diff
@@ -77,15 +77,21 @@ trait Files
             return new Photo($this, $media, $protected);
         }
-        if ($media['_'] !== 'messageMediaDocument') {
+        if ($media['_'] !== 'messageMediaDocument' && $media['_'] !== 'decryptedMessageMediaDocument' && $media['_'] !== 'decryptedMessageMediaExternalDocument') {
```
This does not consider all of the media types for all layers: https://core.telegram.org/type/DecryptedMessageMedia

mtalaeii replied:
its marked as Todo and depend on closing #1415

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Merge branch 'danog:v8' into v8

mtalaeii committed:
Fix
- move service messages of secret chat out of secret namespace and use some of implemented instead in some case
- tag medias as todo
- other fixes
- Run composer cs-fix

mtalaeii committed:
Add phpDoc for DialogDeleteMessages class

danog committed:
Merge branch 'v8' into secret

danog committed:
Fixes

danog committed:
Refactoring

danog committed:
Media support

danog committed:
Fixes

danog committed:
Fixes

danog committed:
fixes

danog committed:
Fixes

danog committed:
Fixes

danog committed:
Fix everything
